### ISSUE TITLE
Scaling on HIDPI screen

### ISSUE THREAD

**Issue by glyh** [HumbleUI/HumbleUI#85]
Using humble UI https://github.com/HumbleUI/HumbleUI/commit/9e71cd32adcc9e82ab33ec538095a39215123f41. 
Here's my hello world result, the text is too small. I'm on Linux Wayland Hyprland.
https://github.com/glyh/javman/tree/7443d789c307cc3f75c6bcb4c14a4f01d1569b26
![image](https://github.com/HumbleUI/HumbleUI/assets/12870742/cac35966-28a4-4de2-b3d9-3815a9b45652)

**Comment by tonsky** [comment]
Is it Linux? How do you set scaling?

**Comment by glyh** [comment]
I don't know how to set scaling. I expect it to "just work". 

Usually it's toolkit's work to provide facilities to configure scaling. For example both QT and GTK provides environment variables for user to set. Here's [an article](https://wiki.archlinux.org/title/HiDPI) for further information for HiDPI scaling on Linux

I also notice the text is small on my end on flutter as well. So I guess this might be upper streamed to either [skija](https://github.com/JetBrains/skija) or [JWM](https://github.com/HumbleUI/JWM)?

**Comment by tonsky** [comment]
I mean, how do you set it in your WM? Which WM do you use? Which distro, too?

**Comment by glyh** [comment]
```
Hyprland, built from branch HEAD at commit c5e28ebcfe00a510922779b2c568cfa52a317445  (props: bump ver 0.37.1).
Date: Sat Mar 16 14:51:49 2024
Tag: v0.37.1

flags: (if any)
```

```
Linux mechrevo 6.8.1-arch1-1 #1 SMP PREEMPT_DYNAMIC Sat, 16 Mar 2024 17:15:35 +0000 x86_64 GNU/Linux
```
I didn't set any specific scaling flags for humbleUI, I have a HiDPI screen. The WM doesn't take care of scaling so I need to have a way to set scaling myself. For example using environment variables for QT/GTK.

Or is it something that has to be done in the codebase using humbleUI?

**Comment by glyh** [comment]
I tend to think this should be something that programmers solve. Closing this.

**Comment by tonsky** [comment]
No-no-no, this should be handled by JWM. The problem is, there is multiple ways to change dpi scaling on Linux. Ideally we should support all of them. So please, let me know yours

**Comment by glyh** [comment]
I don't know what exactly are you looking for but here's how scaling works for different apps on my machine: 

Xresources:
```
Xft.dpi: 165
```

For steam I have 
`/usr/bin/steam-runtime -forcedesktopscaling 1.6 `

For terminals I just the fonts to bigger ones directly. 

For GTK I used to have GDK_SCALE but I don't use many GTK  apps so I deleted them.
For QT I have 
```fish
set -gx QT_SCALE_FACTOR_ROUNDING_POLICY RoundPreferFloor
```

So this is mostly toolkit related. There's no one way to "set scaling". I think one possible way is just to add a new environment variable just for humble ui, like `HUMBLE_UI_SCALE`.

**Comment by glyh** [comment]
I'm learning flutter as well and the hello world UI is as small as the humble UI ones, and they seem to have solution in [developers' code](https://stackoverflow.com/questions/55256762/flutter-is-it-possible-to-auto-scale-whole-ui-based-on-device-size)

**Comment by refset** [comment]
As a temporary workaround, is there some quick & easy way to make everything in the demo app 200% bigger? :slightly_smiling_face: 

(or likely much more complex workaround: the demo could be extended with zoom in/out controls)

**Comment by tonsky** [comment]
![Screenshot 2024-07-23 at 01 51 05](https://github.com/user-attachments/assets/d00cce49-2c25-49a4-827b-e31f7f78843f)
This seem to work. Changing scale manually before initializing default-theme

**Comment by tonsky** [comment]
I just added :scale option to theme, so you can pass it directly as option in a map instead of wrapping in `with-context-classic`
